daemon/cmd/daemon.go

```diff
@@ -406,9 +407,20 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	}
 	// Stop all endpoints (its goroutines) on exit.
 	cleaner.cleanupFuncs.Add(func() {
-		for _, ep := range d.endpointManager.GetEndpoints() {
-			ep.Stop()
+		var wg sync.WaitGroup
```

Comment:
Might be worth adding a Info message here informing the user that Cilium is waiting for the endpoints' go routines to be stopped.